addons/binding/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/handler/NukiSmartLockHandler.java

```diff
@@ -0,0 +1,106 @@
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.nuki.NukiBindingConstants;
+import org.openhab.binding.nuki.dataexchange.BridgeLockActionResponse;
+import org.openhab.binding.nuki.dataexchange.BridgeLockStateResponse;
+import org.openhab.binding.nuki.dataexchange.NukiHttpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NukiSmartLockHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Katter - Initial contribution
+ */
+public class NukiSmartLockHandler extends BaseThingHandler {
+
+    private final static Logger logger = LoggerFactory.getLogger(NukiSmartLockHandler.class);
+
+    public NukiSmartLockHandler(Thing thing) {
+        super(thing);
+        logger.trace("Instantiating NukiSmartLockHandler({})", thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("NukiSmartLockHandler:initialize()");
+        String nukiId = (String) this.getConfig().get(NukiBindingConstants.CONFIG_NUKIID);
+        BridgeLockStateResponse bridgeLockStateResponse = new NukiHttpClient(getBridgeConfig())
+                .getBridgeLockState(nukiId);
+        if (bridgeLockStateResponse.getStatus() == 200) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE,
+                    "Nuki Smart Lock is " + bridgeLockStateResponse.getBridgeLockState().getStateName());
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    bridgeLockStateResponse.getMessage());
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("NukiSmartLockHandler:handleCommand({}, {})", channelUID, command);
+        String nukiId = (String) this.getConfig().get(NukiBindingConstants.CONFIG_NUKIID);
+        if (command.equals(RefreshType.REFRESH)) {
+            BridgeLockStateResponse bridgeLockStateResponse = new NukiHttpClient(getBridgeConfig())
+                    .getBridgeLockState(nukiId);
+            if (bridgeLockStateResponse.getStatus() == 200) {
+                updateState(channelUID,
+                        (bridgeLockStateResponse.getBridgeLockState().getState() == 1 ? OnOffType.ON : OnOffType.OFF));
+            } else {
+                logger.error("Could not refresh Nuki Smart Lock[{}]! Message: {}", nukiId,
+                        bridgeLockStateResponse.getMessage());
+            }
+        } else if (channelUID.getId().equals(NukiBindingConstants.CHANNEL_SMARTLOCKOPENCLOSE)
+                && (command.equals(OnOffType.ON) || command.equals(OnOffType.OFF))) {
+            int lockAction = (command.equals(OnOffType.OFF) ? NukiBindingConstants.LOCKACTIONS_UNLOCK
```

Comment:
OnOffType.OFF is an enum you can compare it with ==, but in this case an instanceOf might be more suiting `command  instanceof OnOffType`